### PR TITLE
Bary3h

### LINKED ISSUE #119: Create tests for edge_cases method
## Description
There are plenty of edge cases, all should be subject of a test.

## Acceptance Criteria
- [ ] Comprehensive inventory of edge cases.
- [ ] One test for each.

## Depends On
While not a hard dependency, it might be better to conclude #118 first.

### LINKED ISSUE #120: Add point_from_zone to Bary3H DGGRS
## Description
This method reverses the process implemented by `zone_from_point`. It should be easier, as it returns "only" needs to pass the barycentric coordinates of the zone centre point to the inverse map projection.

## Acceptance Criteria
- [ ] Convert index back to barycentric coordinates on the face
- [ ] Rollback assignment to icosahedron face (might not be necessary)
- [ ] Invoke projection inverse
- [ ] Create appropriate tests

### PR DESCRIPTION
**Pull request by ldesousa**
ToDo [05-06-2026]:
- Add tests for `edge_cases` method
- Create method for index bundling and add tests
- Debug mode
- Method overloading and/or default method argument values 

Steps:
- [ ] Link PR to the issue or kanban board item.
- [ ] Write a list of what was done, preferably by bullet points.
- [ ] Add README documentation of what's done in the PR, if needed.
- [ ] Request review
- [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ] - [ ]

### CODE REVIEW

**Comment by ldesousa** [comment]
Moving this one to ready. Otherwise it will never be merged, there are always things missing, needing improvement.

File: gp-proj/src/utils/shape.rs [hunk 1]
```diff
@@ -130,3 +130,43 @@ pub fn triangle(
...
     }
 }
+
+/// Convert 2D Cartesian coordinates to Barycentric coordinates
+/// relative to a triangle defined by vertices a, b, c.
+///
+/// Returns (u, v, w) where u + v + w = 1.0
+/// Point is inside the triangle if all three are in [0, 1].
+pub fn cartesian_to_barycentric(
+    p: (f64, f64),
```
Comment (joaocsmanuel) [inline gp-proj/src/utils/shape.rs:140]: There is already a function that does in [github.com/GeoPlegma/GeoPlegma/blob/master/gp-proj/src/projections/polyhedron/spherical_geometry.rs#L254](here). It still needs to be reviewed. So I would leave this here for now.

**Review by joaocsmanuel — APPROVED** [review]
I will just approve so other people can start taking a look.

**Comment by ldesousa** [comment]
This work can only continue when #159 is resolved.

**Comment by ldesousa** [comment]
@joaocsmanuel If the tests pass this can (finally!) be merged.
